libraries/Wire/src/Wire.cpp

```diff
@@ -377,6 +411,37 @@ void TwoWire::onRequest( void (*function)(void) )
   user_onRequest = function;
 }
 
+/**
+  * @brief  Change the size of the buffer.
+  * @param  buffer: pointer to the allocated buffer
+  * @param  length: number of bytes to allocate
+  * @retval pointer to the new buffer location
+  */
+uint8_t *TwoWire::allocateBuffer(uint8_t *buffer, size_t length)
+{
+  // By default we allocate BUFFER_LENGTH bytes. It is the min size of the buffer.
+  if(length < BUFFER_LENGTH) {
+    length = BUFFER_LENGTH;
+  }
+
+  buffer = (uint8_t *)realloc(buffer, length * sizeof(uint8_t));
+  return buffer;
+}
+
+/**
+  * @brief  Reset the buffer. Reduce is size if greater than BUFFER_LENGTH.
```

Comment:
typo: is size / its size
comment says that it reduce the size only if greater than BUFFER_LENGHT but code seems to do it unconditionnally

Reply:
Change in the enhancement